collections/_posts/2023-07-06-Mean_teachers_are_better_role_models.md

```diff
@@ -0,0 +1,203 @@
+
+Limitations of $\Gamma$ model :
+- the model itself generates targets, which may be incorrect
+- the cost of inconsistency outweighs that of misclassification, preventing the learning of new information
+- can lead to a **confirmation bias**
+
+&rarr; Confirmation biais can be mitigated by improving the targets. One way is illustrated with the $\Pi$ model, selecting a different teacher model than the student model
+
+> **NOTE :** The authors mention another way : "to choose the perturbation
+of the representations carefully instead of barely applying additive or multiplicative noise" wich is the subject of another paper of [Miyato et al. 2017](https://doi.org/10.48550/arXiv.1704.03976)[2] that will be refered as VAT+EntMin latter. 
+
+### **The $\Pi$ model [3]**
+- **noise** is added to the model **during inference**
+- result in a "noisy teacher" that can provide **more accurate targets**. 
+
+&rarr; The $\Pi$ model can be further improved by Temporal Ensembling (TE)
+
+### **Temporal Ensembling (TE)**
+- **exponential moving average (EMA)** prediction for each training example is formed by combining the **current version of the model's predictions** with the **predictions made by earlier versions of the model** that evaluated the same example. 
+- improves the quality of the prediction 
+- better results when those predictions are used as the teacher predictions
+
+Limitation of Temporal ensembling :
+- Each **target** (prediction) is **updated only once per training epoch**.
+- The learned information is **incorporated** into the training process at a relatively **slow pace**.
+- The **larger the dataset**, the **longer it takes** for the updates to span the entire dataset.
+
+&rarr; The authors propose the Mean Teacher model that **averages model weights**, instead of predictions like TE, to improve performances of semi-supervised model.
+
+![](/collections/images/mean_teachers/binary_classification_2.jpg)
+
+A sketch of a binary classification task with two labeled examples (large black dots) and one unlabeled example, demonstrating how the choice of the unlabeled target (blue circle) affects the fitted function (gray curve).
+
+(a) A model with no regularization &rarr; free to fit any function that predicts the labeled training examples well  
+(b) A model trained with noisy labeled data (small dots) &rarr; consistent predictions around labeled data points  
+(c) The teacher model (gray curve) is first fitted to the labeled examples, and then left unchanged during the training of the student model. &rarr; Consistency to noise around unlabeled examples provides additional smoothing  
+(d) Noise on the teacher model reduces the bias of the targets without additional training. &rarr; The expected direction of stochastic gradient descent is towards the mean (large blue circle) of individual noisy targets (small blue circles)  
+(e) Ensemble model &rarr; gives an even better expected target
```

Comment:
Markdown does not support alphabetical lists, so we kind of have to work around that. But I tested things locally, and I think it's much easier to read this section if it is formatted as a normal list:
```suggestion
- (a) A model with no regularization &rarr; free to fit any function that predicts the labeled training examples well  
- (b) A model trained with noisy labeled data (small dots) &rarr; consistent predictions around labeled data points  
- (c) The teacher model (gray curve) is first fitted to the labeled examples, and then left unchanged during the training of the student model. &rarr; Consistency to noise around unlabeled examples provides additional smoothing  
- (d) Noise on the teacher model reduces the bias of the targets without additional training. &rarr; The expected direction of stochastic gradient descent is towards the mean (large blue circle) of individual noisy targets (small blue circles)  
- (e) Ensemble model &rarr; gives an even better expected target
```